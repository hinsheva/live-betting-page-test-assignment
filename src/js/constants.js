export const LIVE_MATCHES_API =
  'http://api.unicdn.net/v1/feeds/sportsbook/event/live.jsonp?app_id=ca7871d7&app_key=5371c125b8d99c8f6b5ff9a12de8b85a';
export const LIVE_BETTING_URL = 'https://www.unibet.com/betting#/event/live/';
export const MAX_CACHE_TIME = 120000;
export const PLACE_BET_BUTTON_TEXT = 'Place a bet';
export const LIVE_MATCHES_H1_TEXT = 'Live matches';
export const LIVE_MATCHES_P_TEXT = 'Here is a list of matches that are live right now.';
export const FETCH_MATCHES_ERROR_MESSAGE =
  'Match data is not available now. Please Try again later.';
