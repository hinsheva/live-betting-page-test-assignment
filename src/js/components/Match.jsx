import React from 'react';
import { LIVE_BETTING_URL, ACTIONS } from '../constants';
import { formatDate, getSportIcon } from '../utils/utils';

const Match = ({ liveEvent, send }) => {
  const { start, awayName, homeName, id, sport, awayScore, homeScore } = liveEvent;

  const eventDate = formatDate(start);
  const sportIcon = getSportIcon(sport);
  const liveEventUrl = `${LIVE_BETTING_URL}${id}`;

  return (
    <div className="match">
      <div className="score">
        {awayScore && homeScore ? `${awayScore} - ${homeScore}` : 'No score'}
      </div>
      <div className="team-container">
        <img src={sportIcon} alt={sport || 'sport'} /> {awayName} {awayName && homeName ? '-' : ''}{' '}
        {homeName}
      </div>
      <div className="date">{eventDate}</div>
      <button className="bet-button" onClick={() => send(ACTIONS.PLACE_BET, { liveEventUrl })}>
        {' '}
        Place a bet
      </button>
    </div>
  );
};

export default Match;
