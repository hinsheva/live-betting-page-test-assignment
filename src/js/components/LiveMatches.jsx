// @flow

import React from 'react';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Slider from 'react-slick';
import SideBlock from './SideBlock';
import Match from './Match';
import Error from './Error';
import Loading from './Loading';
import { mappedLiveEventsData, carouselStyle, getLiveEventsFromCache } from '../utils/utils';
import { STATES } from '../constants';

const LiveMatches = ({ current, send }: { current: string, send: Function }) => {
  const { liveEvents, error } = current.context;
  const { LOADING, LOADED, FAILED } = STATES;

  getLiveEventsFromCache({ send });

  return (
    <div id="content">
      <article>
        <h1>Live matches</h1>
        <p className="preamble">Here is a list of matches that are live right now.</p>
        <SideBlock />
        <div id="live-matches">
          <Slider {...carouselStyle}>
            {current.matches(LOADING) && <Loading />}
            {current.matches(LOADED) &&
              mappedLiveEventsData(liveEvents).map((match) => (
                <Match key={match.id} liveEvent={match} send={send} />
              ))}
            {current.matches(FAILED) && <Error error={error} />}
          </Slider>
        </div>
      </article>
    </div>
  );
};

export default LiveMatches;
