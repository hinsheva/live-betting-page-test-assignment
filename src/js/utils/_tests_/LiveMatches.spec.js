import React from 'react';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import LiveMatches from '../../components/LiveMatches';
import Match from '../../components/Match';
import { LIVE_MATCHES_H1_TEXT } from '../../constants';

describe('LiveMatches', () => {
  const props = {
    current: {
      context: {
        liveEvents: [],
      },
      matches: Function,
    },
    send: Function,
  };

  const matchProps = {
    send: Function,
    liveEvent: {
      homeName: 'Test Team #1',
      awayName: 'Test Team #2',
      sport: "some sport that we don't have icon for",
      start: 'Today, 18:20',
      homeScore: '7',
      awayScore: '2',
    },
  };

  const renderer = TestRenderer.create(<LiveMatches {...props} />);
  const instance = renderer.root;

  it('should render "Place a bet" button', () => {
    const element = instance.findByType('h1');
    expect(element.props.children).toEqual(LIVE_MATCHES_H1_TEXT);
  });

  it('should render LiveMatches component with children', () => {
    const renderedTree = renderer.toJSON();
    expect(renderedTree.children.length).toBe(1);
  });

  it('should render LiveMatches component with children', () => {
    const renderedTree = renderer.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it('renders correctly child component Match', () => {
    const childRenderer = new ShallowRenderer();
    childRenderer.render(<Match {...matchProps} />);
    const result = childRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
