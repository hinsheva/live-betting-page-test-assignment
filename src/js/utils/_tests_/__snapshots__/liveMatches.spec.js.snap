// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveMatches renders correctly child component Match 1`] = `
<div
  className="match"
>
  <div
    className="score"
  >
    2 - 7
  </div>
  <div
    className="team-container"
  >
    <img
      alt="some sport that we don't have icon for"
      src="defaultIcon.png"
    />
     
    Test Team #2
     
    -
     
    Test Team #1
  </div>
  <div
    className="date"
  >
    No date
  </div>
  <Button
    liveEventUrl="https://www.unibet.com/betting#/event/live/undefined"
    send={[Function]}
  />
</div>
`;

exports[`LiveMatches should render LiveMatches component with children 1`] = `
<div
  id="content"
>
  <article>
    <h1>
      Live matches
    </h1>
    <p
      className="preamble"
    >
      Here is a list of matches that are live right now.
    </p>
    <aside>
      <h2>
        Live betting
      </h2>
      <p>
        Place your bets as the action unfolds. We offer a wide selection of live betting events and you can place both single and combination bets.
      </p>
      <p>
        You will be able to see an in-play scoreboard with the current result and match stats, while on selected events you will also be able to watch the action live with Unibet TV on the desktop site.
      </p>
    </aside>
    <div
      id="live-matches"
    >
      <div
        className="slick-slider slick-initialized"
        dir="ltr"
      >
        <button
          className="slick-arrow slick-prev"
          data-role="none"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
          type="button"
        >
           
          Previous
        </button>
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-1}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="error"
                >
                  <div />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="loading"
                >
                  <div>
                    Loading...
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="error"
                >
                  <div />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="loading"
                >
                  <div>
                    Loading...
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="error"
                >
                  <div />
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          className="slick-arrow slick-next"
          data-role="none"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
          type="button"
        >
           
          Next
        </button>
      </div>
    </div>
  </article>
</div>
`;
